src/components/review/ProposalTechnicalReview.tsx

```diff
@@ -1,64 +1,121 @@
   id: number;
+  setFormDirty: (dirty: boolean) => void;
 }) {
   const { api } = useDataApiWithFeedback();
+  const [shouldSubmit, setShouldSubmit] = useState(false);
+  const classes = useStyles();
 
-  const initialValues = {
+  const initialValues: TechnicalReviewFormType = {
     status: props?.data?.status || '',
     timeAllocation: props?.data?.timeAllocation || '',
     comment: props?.data?.comment || '',
     publicComment: props?.data?.publicComment || '',
   };
 
+  const PromptIfDirty = () => {
+    const formik = useFormikContext();
+
+    return (
+      <Prompt
+        when={formik.dirty && formik.submitCount === 0}
+        message="Changes you recently made in this tab will be lost! Are you sure?"
+      />
+    );
+  };
+
+  const handleUpdateOrSubmit = async (
+    values: TechnicalReviewFormType,
+    method: 'submitTechnicalReview' | 'addTechnicalReview'
+  ) => {
+    const shouldSubmit = method === 'submitTechnicalReview';
+    const successMessage = `Technical review ${
+      shouldSubmit ? 'submitted' : 'updated'
+    } successfully!`;
+
+    const result = await api(successMessage)[method]({
+      proposalID: props.id,
+      timeAllocation: +values.timeAllocation,
+      comment: values.comment,
+      publicComment: values.publicComment,
+      status: TechnicalReviewStatus[values.status as TechnicalReviewStatus],
+      submitted: shouldSubmit,
+    });
+
+    if (!(result as any)[method].error) {
+      props.setReview({
+        proposalID: props?.data?.proposalID,
+        timeAllocation: +values.timeAllocation,
+        comment: values.comment,
+        publicComment: values.publicComment,
+        status: TechnicalReviewStatus[values.status as TechnicalReviewStatus],
+        submitted: shouldSubmit,
+      } as CoreTechnicalReviewFragment);
+    }
+  };
+
   return (
-    <Fragment>
+    <>
       <Typography variant="h6" gutterBottom>
         Technical Review
       </Typography>
       <Formik
         initialValues={initialValues}
         validationSchema={proposalTechnicalReviewValidationSchema}
         onSubmit={async (values): Promise<void> => {
-          await api(
-            'Technical review updated successfully!'
-          ).addTechnicalReview({
-            proposalID: props.id,
-            timeAllocation: +values.timeAllocation,
-            comment: values.comment,
-            publicComment: values.publicComment,
-            status:
-              TechnicalReviewStatus[values.status as TechnicalReviewStatus],
-          });
-          props.setReview({
-            proposalID: props?.data?.proposalID,
-            timeAllocation: +values.timeAllocation,
-            comment: values.comment,
-            publicComment: values.publicComment,
-            status:
-              TechnicalReviewStatus[values.status as TechnicalReviewStatus],
-          } as CoreTechnicalReviewFragment);
+          if (shouldSubmit) {
+            const confirmed = window.confirm(
```

Comment:
I think window.confirm() is sufficient here, but just wanted to note that there is also a HOC 
 withConfirm https://github.com/UserOfficeProject/user-office-frontend/blob/develop/src/utils/withConfirm.tsx

Reply:
I was thinking about this but I know that this technical review thing might change because we mentioned that we like the full screen modal sliding from bottom to top like we have for SEP reviews. If this is the case I know that you mentioned once it is a bit hard to make this confirm on top of another modal but if I am wrong I can try to change it and make it work.

Reply:
I fixed it for now and it uses `withConfirm` HOC I think it is better like this at the moment thanks.